src/PageLayout/useStickyPaneHeight.ts

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import {useInView} from 'react-intersection-observer'
+
+/**
+ * Calculates the height of the sticky pane such that it always
+ * fits into the viewport even when the header or footer are visible.
+ */
+// TODO: Handle sticky header
+export function useStickyPaneHeight() {
+  // Default the height to the viewport height
+  const [height, setHeight] = React.useState('100vh')
+
+  // Create intersection observers to track the top and bottom of the content region
+  const [contentTopRef, contentTopInView, contentTopEntry] = useInView()
+  const [contentBottomRef, contentBottomInView, contentBottomEntry] = useInView()
+
+  // Calculate the height of the sticky pane based on the position of the
+  // top and bottom of the content region
+  const calculateHeight = React.useCallback(() => {
+    // Uncomment to debug
+    // console.log('Recalculating pane height...')
+
+    const topRect = contentTopEntry?.target.getBoundingClientRect()
+    const bottomRect = contentBottomEntry?.target.getBoundingClientRect()
+
+    const topOffset = topRect ? Math.max(topRect.top, 0) : 0
+    const bottomOffset = bottomRect ? Math.max(window.innerHeight - bottomRect.bottom, 0) : 0
+
+    // Safari's elastic scroll feature allows you to scroll beyond the scroll height of the page.
+    // We need to account for this when calculating the offset.
+    const overflowScroll = Math.max(window.scrollY + window.innerHeight - document.body.scrollHeight, 0)
```

Comment:
Do you think this should be calculated on the scrollcontainer? Or is the `PageLayout` always meant to be used as a top level component?

Reply:
PageLayout should always be the top-level component. But this is a good consideration. I'll keep it as `window` for now and adjust it if needed.

Reply:
Actually, it looks like memex uses PageLayout inside a scroll container that isn't `document.body`. I'll fix this in this PR

Reply:
> PageLayout should always be the top-level component.

Can you say more about this, would the top level navigation be inside PageLayout.Header? 

Right now, I think issues has top level navigation and page layout as siblings, so there's some margin on top

Reply:
Yeah, I was wrong about PageLayout always being top-level. @joshblack and I just updated the implementation to work if PageLayout is nested in a scroll container: https://github.com/primer/react/pull/2212/commits/52e174fd535c5e29c552e685003a3cac7c578137